### PR TITLE
feat(api): finish document publishing

### PR DESCRIPTION
**Pull request by chrisdalykinandcarta**
## Describe your changes

Implement file publishing

## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [ ] I have performed a self-review of my own code
- [ ] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [ ] I have referenced the ticket number above
- [ ] I have provided a description of the ticket
- [ ] I have included unit tests to cover any testable code changes

### CODE REVIEW

**Review by francescobrenci-AR — DISMISSED** [review]

File: apps/functions/document-publish/publish-document/index.js [hunk 1]
```diff
@@ -1,6 +1,57 @@
...
+	context.log(`Deploying to file name ${publishFileName}`);
+
+	await blobClient.copyFile({
+		sourceContainerName: config.BLOB_SOURCE_CONTAINER,
+		sourceBlobName: documentURI,
+		destinationContainerName: config.BLOB_PUBLISH_CONTAINER,
+		destinationBlobName: publishFileName
+	});
+
+	await got
```
Comment (richmoh) [inline apps/functions/document-publish/publish-document/index.js:34]: @chrisdalykinandcarta What would happen if this request failed for whatever reason? Should we consider wrapping in a try catch and reversing any changes made above or does it not matter?

File: apps/api/src/server/applications/documents/documents.service.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ * @param {string} newStatus
+ */
+export const updateStatus = async (guid, newStatus) => {
+	const document = await documentRepository.getByDocumentGUID(guid);
+
+	const currentDocumentStatus = document?.documentVersion[0]?.publishedStatus;
+
+	/** @type {string} */
+	// @ts-ignore
+	const status = nextStatusInDocumentStateMachine(currentDocumentStatus, newStatus);
```
Comment (richmoh) [inline apps/api/src/server/applications/documents/documents.service.js:20]: call `nextStatus` ?

File: apps/functions/document-publish/publish-document/index.js [hunk 3]
```diff
@@ -1,6 +1,57 @@
+import config from './config';
```
Comment (beejjacobs) [inline apps/functions/document-publish/publish-document/index.js:1]: Given the issue I had, this might need `.js`

**Review by richmoh — DISMISSED** [review]
Nice. Approved with a couple of small comments.

File: packages/blob-storage-client/src/blob-storage-client.js [hunk 4]
```diff
@@ -144,20 +143,19 @@ export class BlobStorageClient {
 
 	/**
 	 *
-	 * @param {{currentContainer: string, currentFilePath: string, desiredContainer: string, desiredFilePath: string}} blobStorageHost
+	 * @param {{sourceContainerName: string, sourceBlobName: string, destinationContainerName: string, destinationBlobName: string}} blobStorageHost
```
Comment (beejjacobs) [inline packages/blob-storage-client/src/blob-storage-client.js:146]: These are better names!

**Review by beejjacobs — APPROVED** [review]
Not much to add, looks good.

File: apps/api/src/server/applications/documents/documents.service.js (hunk 2 above)
Comment (chrisdalykinandcarta) [reply apps/api/src/server/applications/documents/documents.service.js:20]: It's a good call, but it's passed as a named parameter on the next line and the parameter is called 'status', so just keeping consistent with the repository parameter (but also so I don't have to rename it when passing)

File: apps/functions/document-publish/publish-document/index.js (hunk 1 above)
Comment (chrisdalykinandcarta) [reply apps/functions/document-publish/publish-document/index.js:34]: Good catch (pun intended), we don't really have a pattern for this at the moment - but if the request fails, we're probably going to assume it's a transient error and bubbling the exception back up to the function runtime will attempt a retry with exponential back-off.

There will potentially be cases where we hit issues we'll want to step back from, just not sure what these are yet. The function itself has to be idempotent so we probably do want to figure out what happens if a document is already in a certain state.

File: apps/functions/document-publish/publish-document/index.js (hunk 3 above)
Comment (chrisdalykinandcarta) [reply apps/functions/document-publish/publish-document/index.js:1]: Good spot!

**Review by francescobrenci-AR — APPROVED** [review]

**Pull request merged**